From e891eae105ba3dff9adabd602f9d8f195e028065 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Barnab=C3=A1s=20P=C5=91cze?= <pobrn@protonmail.com>
Date: Sat, 17 Oct 2026 15:42:15 +0200
Subject: [PATCH] spa: debug: buffer: add size checks

Don't print metadata items if the size is not big enough.

diff --git a/spa/include/spa/debug/buffer.h b/spa/include/spa/debug/buffer.h
index c0b533564..16e78e57e 100644
--- a/spa/include/spa/debug/buffer.h
+++ b/spa/include/spa/debug/buffer.h
@@ -49,6 +49,9 @@ SPA_API_DEBUG_BUFFER int spa_debugc_buffer(struct spa_debug_context *ctx, int in
 		case SPA_META_Header:
 		{
 			struct spa_meta_header *h = (struct spa_meta_header*)m->data;
+			if (m->size < sizeof(*h))
+				break;
+
 			spa_debugc(ctx, "%*s" "    struct spa_meta_header:", indent, "");
 			spa_debugc(ctx, "%*s" "      flags:      %08x", indent, "", h->flags);
 			spa_debugc(ctx, "%*s" "      offset:     %u", indent, "", h->offset);
@@ -60,6 +63,9 @@ SPA_API_DEBUG_BUFFER int spa_debugc_buffer(struct spa_debug_context *ctx, int in
 		case SPA_META_VideoCrop:
 		{
 			struct spa_meta_region *h = (struct spa_meta_region*)m->data;
+			if (m->size < sizeof(*h))
+				break;
+
 			spa_debugc(ctx, "%*s" "    struct spa_meta_region:", indent, "");
 			spa_debugc(ctx, "%*s" "      x:      %d", indent, "", h->region.position.x);
 			spa_debugc(ctx, "%*s" "      y:      %d", indent, "", h->region.position.y);

